Sources/StreamVideo/WebRTC/v2/PeerConnection/Extensions/RTCRtpEncodingParameters+Convenience.swift

```diff
@@ -17,12 +17,19 @@ extension RTCRtpEncodingParameters {
     /// - Note: The `rid` (Restriction Identifier) is set to the codec's quality.
     ///         The `maxBitrateBps` is set to the codec's maximum bitrate.
     ///         If the codec has a `scaleDownFactor`, it's applied to `scaleResolutionDownBy`.
-    convenience init(_ codec: VideoCodec) {
+    convenience init(
+        _ layer: VideoLayer,
+        preferredVideoCodec: VideoCodec?
+    ) {
         self.init()
-        rid = codec.quality
-        maxBitrateBps = (codec.maxBitrate) as NSNumber
-        if let scaleDownFactor = codec.scaleDownFactor {
-            scaleResolutionDownBy = (scaleDownFactor) as NSNumber
+        rid = layer.quality.rawValue
+        maxBitrateBps = (layer.maxBitrate) as NSNumber
+        if preferredVideoCodec?.isSVC == true {
+            scalabilityMode = "L3T2_KEY"
```

Comment:
what's this? Some value we have to send for the scalability mode?

Reply:
Yeah, this is straight from the [Spec](https://www.notion.so/stream-wiki/SVC-Codecs-VP9-AV1-and-Dynamic-Simulcast-layer-toggling-1226a5d7f9f680168677c54e1d547600?pvs=4#1226a5d7f9f680acb353f36dfd0ba951). This is starting scalabilityMode we set (3 layers 2 Tracks with Keyframe). However, later on the SFU can send any value from [here](https://www.w3.org/TR/webrtc-svc/#scalabilitymodes*). I thought to make an enum for type-safety but it seems unneccessary. WDYT?

Reply:
no, that's fine, it was just out of curiosity.